src/Raven.Server/Monitoring/OpenTelemetry/MetricsManager.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Util;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Monitoring.OpenTelemetry;
+
+public class MetricsManager
+{
+    private readonly RavenServer _server;
+    private readonly SemaphoreSlim _locker = new(1, 1);
+    private ServerMetrics _serverMetrics;
+    private readonly Dictionary<string, DatabaseWideMetrics> _loadedDatabases = new(StringComparer.OrdinalIgnoreCase);
+
+    public MetricsManager(RavenServer server)
+    {
+        _server = server;
+        _server.ServerStore.LicenseManager.LicenseChanged += OnLicenseChanged;
+    }
+
+    private void OnLicenseChanged()
+    {
+        if (_server.Configuration.Monitoring.OpenTelemetry.Enabled == false)
+            return;
+
+        _locker.Wait();
+        try
+        {
+            var activate = _server.ServerStore.LicenseManager.CanUseOpenTelemetryMonitoring(withNotification: true);
+            if (activate)
+            {
+                Execute();
+                //turn on
+            }
+            else
+            {
+                //when we dispose meter all registered thing will die eventually however metrics still be exported but there will be no instruments
+            }
+        }
+        catch (ObjectDisposedException)
+        {
+            // ignore
+            // we are shutting down the server
+        }
+        finally
+        {
+            _locker.Release();
+        }
+    }
+
+    public void Execute()
+    {
+        if (_server.Configuration.Monitoring.OpenTelemetry.Enabled == false)
+            return;
+
+        _locker.Wait();
+
+        try
+        {
+            var activate = _server.ServerStore.LicenseManager.CanUseOpenTelemetryMonitoring(withNotification: true);
+            if (activate)
+            {
+                if (_serverMetrics != null)
+                    throw new InvalidOperationException("Cannot start SNMP Engine because it is already activated. Should not happen!");
+
+                RegisterServerWideMeters();
+            }
+
+            _server.ServerStore.DatabasesLandlord.OnDatabaseLoaded += AddDatabasesIfNecessary;
+        }
+        catch (Exception e)
+        {
+            Debugger.Launch();
+            Debugger.Break();
+            throw;
+        }
+        finally
+        {
+            _locker.Release();
+        }
+
+        AsyncHelpers.RunSync(AddDatabases);
+    }
+
+    private async Task AddDatabases()
+    {
+        if (_server.Configuration.Monitoring.OpenTelemetry.Databases == false)
+            return;
+
+        await _locker.WaitAsync();
```

Comment:
beside this, I do not see any async code here and IIRC SemaphoreSlim has sync version of that. Do we want to take leverage of it?